src/main/java/io/vertx/core/net/impl/pool/CombinerExecutor.java

```diff
@@ -12,61 +12,149 @@
 
 import io.netty.util.internal.PlatformDependent;
 
+import java.time.Duration;
+import java.util.Objects;
 import java.util.Queue;
+import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.LongSupplier;
 
 /**
  * Lock free executor.
- *
+ * <p>
  * When a thread submits an action, it will enqueue the action to execute and then try to acquire
  * a lock. When the lock is acquired it will execute all the tasks in the queue until empty and then
  * release the lock.
  */
 public class CombinerExecutor<S> implements Executor<S> {
+  @FunctionalInterface
+  public interface YieldCondition {
```

Comment:
for now as this is only used in the same package, make this package private

Reply:
But if it becomes part of the submitAndContinue API, will need it to be public, as `Continuation` is

Reply:
for now this API is only for SimpleConnectionPool. It is decoupled from the connection pool only for purpose of testing and benchmarking.